arbor/threading/threading.cpp

```diff
@@ -129,31 +152,25 @@ task_system::task_system(int nthreads): count_(nthreads), q_(nthreads) {
 }
 
 task_system::~task_system() {
+    current_task_priority_ = -1;
```

Comment:
Shouldn't we wait until all the threads have quit and been joined before resetting the `current_task_priority_` and `current_task_queue_`? It shouldn't be a big deal because we're destructing everything, but I'm not entirely sure that this won't cause some subtle bug. Maybe I'm being too cautious though...

Reply:
Because they are thread local, it will only reset the values in the thread calling the destructor, which won't be taking on any more jobs once the destruction has started. The only advantage though in setting them early would be to avoid problems should anything below throw, but the destructor is implicit noexcept anyway, so an exception here would terminate regardless.

Reply:
That makes sense, thanks!